File: 061/main.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {ListNode}
 */

function ListNode (val) {
  this.val = val
  this.next = null
}

var rotateRight = function (head, k) {
  if (head == null) return head
  let num = 1
  let flag = head
  while (flag.next != null) {
    flag = flag.next
    num++
  }
  flag.next = head
  let i = 0
  k = k % num
  while (i < num - k) {
    head = head.next
    i++
  }
  const res = new ListNode(-1)
  let tmp = res
  i = 1
  while (i < num) {
    tmp.next = new ListNode(-1)
    tmp.val = head.val
    head = head.next
    tmp = tmp.next
    i++
  }
  tmp.val = head.val
  return res
}

const list = {
  val: 1,
  next: {
    val: 2,
    next: {
      val: 3,
      next: {
        val: 4,
        next: {
          val: 5,
          next: null
        }
      }
    }
  }
}
console.log(rotateRight(list, 2))

File: 061/main.cpp
#include <iostream>

using namespace std;

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if (head == NULL || k == 0) return head;
        int sum = 1;
        ListNode* ll = head;
        while (ll->next != NULL) {
            ll = ll->next;
            sum++;
        }
        ll->next = head;
        int m = k % sum;
        for (int i = 0; i < sum - m; i++) {
            ll = ll->next;
        }
        head = ll->next;
        ll->next = NULL;
        return head;
    }
};

int main()
{
    Solution sol;
    ListNode* head = new ListNode(1);
    ListNode* ll;
    return 0;
}
